modules/storages/app/controllers/storages/admin/storages_controller.rb

```diff
@@ -86,12 +86,13 @@ def new
   def create
     service_result = Storages::Storages::CreateService.new(user: current_user).call(permitted_storage_params)
     @object = service_result.result
+    @oauth_application = result_from_first_dependent_service(service_result)
 
-    if service_result.success? && (@oauth_application = service_result.dependent_results&.first&.result)
+    if service_result.success? && @oauth_application
       flash[:notice] = I18n.t(:notice_successful_create)
       render :show_oauth_application
     else
-      @errors = service_result.errors
+      @errors = service_result.merge_dependent_errors(@object)
```

Comment:
This is confusing as it took me a while to understand, that `@object` does indeed has a value here. This is a side effect that is not visible here. I suggest returning nil from the `CreateService` if the service call was not successful, as there is also an error attribute defined. Consider using an error object to transport further data for error handling.